core/app/[locale]/(default)/(faceted)/category/[slug]/page.tsx

```diff
@@ -47,97 +68,184 @@ export async function generateMetadata({ params }: Props): Promise<Metadata> {
   };
 }
 
-export default async function Category(props: Props) {
-  const searchParams = await props.searchParams;
-  const params = await props.params;
+async function getBreadcrumbs(categoryId: number): Promise<Breadcrumb[] | null> {
+  const category = await getCategory(categoryId);
 
-  const { locale, slug } = params;
+  return removeEdgesAndNodes(category.breadcrumbs).map(({ name, path }) => ({
+    label: name,
+    href: path ?? '#',
+  }));
+}
 
-  setRequestLocale(locale);
+async function getTitle(categoryId: number): Promise<string | null> {
+  const category = await getCategory(categoryId);
 
-  const t = await getTranslations('Category');
+  return category.name;
+}
 
-  const categoryId = Number(slug);
+async function getSearch(categoryId: number, searchParamsPromise: Promise<SearchParams>) {
```

Comment:
We've been wrapping these in `cache` just in case.

Reply:
All of these call `fetchSearchResults`, which is itself cached

Reply:
Ah right! Cool